tap_github/autenticator.py

```diff
@@ -0,0 +1,102 @@
+"""Classes to assist in authenticating to the GitHub API."""
+
+import logging
+from datetime import datetime
+from os import environ
+from random import choice, shuffle
+from typing import Any, Dict, Optional
+
+from singer_sdk.streams import RESTStream
+
+
+class TokenRateLimit:
+    """A class to store token rate limiting information."""
+
+    DEFAULT_RATE_LIMIT = 5000
+    RATE_LIMIT_BUFFER = 1000
```

Comment:
Is this a buffer of quota you don't allow the tap to use? maybe worth a docstring?